core/src/main/java/com/adobe/aem/commons/assetshare/components/actions/share/impl/EmailShareServiceImpl.java

```diff
@@ -230,14 +231,32 @@ private String getSignature(final SlingHttpServletRequest request) throws Reposi
      * @param userData the Map of data provided by the end-user (usually derived from the Request) to use in the email.
      * @return the protected Map; all String's are xss protected for HTML.
      */
-    private Map<String, Object> xssProtectUserData(Map<String, Object> userData) {
-        for(final Map.Entry<String, Object> entry : userData.entrySet()) {
-            if (entry.getValue() instanceof String) {
-                userData.put(entry.getKey(), xssAPi.encodeForHTML((String)entry.getValue()));
+    private Map<String, Object> xssProtectUserData(Map<String, Object> dirtyUserData) {
+    	    
+    	    Map<String, Object> cleanUserData = new HashMap<String, Object>();
+        for(final Map.Entry<String, Object> entry : dirtyUserData.entrySet()) {
+        		
+            if (entry.getValue() instanceof String[]) {
+            		cleanUserData.put(entry.getKey(), xssCleanData((String[])entry.getValue()));
+            }
+            else if (entry.getValue() instanceof String) {
+            	    cleanUserData.put(entry.getKey(), xssCleanData((String)entry.getValue()));
             }
         }
 
-        return userData;
+        return cleanUserData;
+    }
+    
+    private String[] xssCleanData(String[] dirtyData) {
+    	    List<String> cleanValues = new ArrayList<String>();
+		for(String val : dirtyData) {
+			cleanValues.add(xssAPi.encodeForHTML(xssAPi.filterHTML(val)));
```

Comment:
Should take the opportunity rename `xssAPi` variable to proper Java naming convention; no idea how I got this in there - musta been a copy/pasta.